### PR TITLE
initial implementation of 'yb deploy'

### PR DESCRIPTION
**Pull request by ksmskm**
### notes
both folders return one filepath result for: 
`bazel query --noshow_progress "filter('_deploy.apply$', ':all')"`
ex: 
`//nodejs:nodejs-test-cd_deploy.apply`

I'd like to test this against one that has multiple filepaths

### manual test results
`github.com/yourbase/test-cd`
```
COMBINED OUTPUT: Loading:
Loading: 0 packages loaded
INFO: Analysed target //:test_cd_deploy.apply (0 packages loaded).
INFO: Found 1 target...
[0 / 3] [-----] BazelWorkspaceStatusAction stable-status.txt
Target //:test_cd_deploy.apply up-to-date:
  bazel-bin/test_cd_deploy.apply
INFO: Elapsed time: 0.473s, Critical Path: 0.01s
INFO: Build completed successfully, 1 total action

INFO: Running command line: bazel-bin/test_cd_deploy.apply
deployment "test-cd-app-deployment" unchanged
service "test-cd-svc" unchanged
```

`github.com/yourbase/test-cd/nodejs`

```
COMBINED OUTPUT: Loading:
Loading: 0 packages loaded
INFO: Analysed target //nodejs:nodejs-test-cd_deploy.apply (0 packages loaded).
INFO: Found 1 target...
[0 / 2] [-----] BazelWorkspaceStatusAction stable-status.txt
Target //nodejs:nodejs-test-cd_deploy.apply up-to-date:
  bazel-bin/nodejs/nodejs-test-cd_deploy.apply
INFO: Elapsed time: 0.469s, Critical Path: 0.01s
INFO: Build completed successfully, 1 total action

INFO: Running command line: bazel-bin/nodejs/nodejs-test-cd_deploy.apply
deployment "nodejs-test-cd-app-deployment" configured
service "nodejs-test-cd-svc" unchanged
```

---
This change is [Reviewable](https://reviewable.io/reviews/yourbase/yourbase/99)

### CODE REVIEW

File: yb/cmd/deploy.go [hunk 1]
```diff
@@ -15,9 +18,35 @@ var deployCmd = &cobra.Command{
...
+		// get combined output to STDOUT & STDERR
+		output, err = run.CombinedOutput()
+		combinedOutput := string(output)
+
+		// STDOUT is helpful whether or not there is an error
+		fmt.Println("COMBINED OUTPUT:", combinedOutput)
+		if err != nil {
+			fmt.Println("ERR:", err)
+			os.Exit(1)
+		}
```
Comment (kbfastcat) [inline yb/cmd/deploy.go]: Should there be a uniform error message from all yb commands?  @nictuku @ksmskm

**Comment by nictuku** [comment]
Review status: 0 of 1 files reviewed at latest revision, 4 unresolved discussions.

---

*[yb/cmd/deploy.go, line 27 at r1](https://reviewable.io:443/reviews/yourbase/yourbase/99#-L7WptPwNXZ9Gb4TFX9j:-L7WptPwNXZ9Gb4TFX9k:b-11kfwg) ([raw file](https://github.com/yourbase/yourbase/blob/3032ff449f0c5b09b5bdd2eb297884aed06a9d04/yb/cmd/deploy.go#L27)):*
> ```Go
> 		output, err := query.Output()
> 		if err != nil {
> 			fmt.Println(err)
> ```

When printing errors, use fmt.Fprintf(os.Stderr, "message")

---

*[yb/cmd/deploy.go, line 33 at r1](https://reviewable.io:443/reviews/yourbase/yourbase/99#-L7Wq1q5NIJ2h_a5VmTV:-L7Wq1q5NIJ2h_a5VmTW:b27w3mi) ([raw file](https://github.com/yourbase/yourbase/blob/3032ff449f0c5b09b5bdd2eb297884aed06a9d04/yb/cmd/deploy.go#L33)):*
> filepaths = strings.TrimSpace(filepaths)

Reduce the temporary variables. More compact code is more readable, in Go at least.

filepaths := strings.TrimSpace(string(output))

---

*[yb/cmd/deploy.go, line 45 at r1](https://reviewable.io:443/reviews/yourbase/yourbase/99#-L7WqM8qFDCS28DgTeZ-:-L7WqM8qFDCS28DgTeZ0:b-dv6hql) ([raw file](https://github.com/yourbase/yourbase/blob/3032ff449f0c5b09b5bdd2eb297884aed06a9d04/yb/cmd/deploy.go#L45)):*
> ```Go
> 
> 		// STDOUT is helpful whether or not there is an error
> 		fmt.Println("COMBINED OUTPUT:", combinedOutput)
> ```

To reduce the logging noise, don't say "COMBINED OUTPUT". Also, you can also do this in fewer lines.

output, err = run.CombinedOutput()
fmt.Println(output) // or string(output) if the former doesn't work.

---

*[yb/cmd/deploy.go, line 49 at r1](https://reviewable.io:443/reviews/yourbase/yourbase/99#-L7Wpcrgw-nD3uuWUSTk-r1-49:-L7WqgHBglSTWUqBkj0D:b-6kahg0) ([raw file](https://github.com/yourbase/yourbase/blob/3032ff449f0c5b09b5bdd2eb297884aed06a9d04/yb/cmd/deploy.go#L49)):*
<details><summary><i>Previously, kbfastcat (Jeremy Whittemore) wrote…</i></summary><blockquote>

Should there be a uniform error message from all yb commands?  @nictuku @ksmskm 
</blockquote></details>

Yes we should create a utility function to print-an-error-and-exit.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/yourbase/yourbase/99)*

**Comment by nictuku** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/yourbase/yourbase/99)*

**Comment by nictuku** [comment]
LGTM

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[yb/cmd/deploy.go, line 49 at r1](https://reviewable.io:443/reviews/yourbase/yourbase/99#-L7Wpcrgw-nD3uuWUSTk-r1-49:-L7YBx69EZ8NXyyZ23Tx:b-ekt8xe) ([raw file](https://github.com/yourbase/yourbase/blob/3032ff449f0c5b09b5bdd2eb297884aed06a9d04/yb/cmd/deploy.go#L49)):*
<details><summary><i>Previously, nictuku (Yves Junqueira) wrote…</i></summary><blockquote>

Yes we should create a utility function to print-an-error-and-exit.
</blockquote></details>

Let's do it in a later change.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/yourbase/yourbase/99#-:-L7YC14hgQUbeKJNrVvN:bb6ywmx)*

**Comment by nictuku** [comment]
This fixes #45

**Pull request merged**